proto/redpanda/api/dataplane/v1/mcp.proto

```diff
@@ -0,0 +1,664 @@
+      summary: "Start a Redpanda Connect MCP server"
+      description: "Start a stopped Redpanda Connect MCP server."
+      responses: {
+        key: "200"
+        value: {
+          description: "Started"
+          schema: {
+            json_schema: {ref: ".redpanda.api.dataplane.v1.MCPServer"}
+          }
+        }
+      }
+      responses: {
+        key: "404"
+        value: {
+          description: "Not Found"
+          schema: {
+            json_schema: {ref: ".google.rpc.Status"}
+          }
+        }
+      }
+    };
+    option (redpanda.api.auth.v1.authorization) = {
+      required_permission: PERMISSION_EDIT
+      api: API_REDPANDA_MCP_SERVER
+    };
+  }
+
+  // The configuration schema includes available components and processors in this Redpanda Connect MCP Server instance.
+  rpc GetMCPServerServiceConfigSchema(GetMCPServerServiceConfigSchemaRequest) returns (GetMCPServerServiceConfigSchemaResponse) {
+    option (google.api.http) = {get: "/v1/redpanda-connect/mcp-config-schema"};
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+      summary: "Retrieve the schema for Redpanda Connect MCP Servers configurations."
+      description: "The configuration schema includes available components and processors in this Redpanda Connect MCP Server instance."
+      responses: {
+        key: "200"
+        value: {
+          description: "OK"
+          schema: {
+            json_schema: {ref: ".redpanda.api.dataplane.v1.GetMCPServerServiceConfigSchemaResponse"}
+          }
+        }
+      }
+    };
+    option (redpanda.api.auth.v1.authorization) = {
+      required_permission: PERMISSION_VIEW
+      api: API_REDPANDA_MCP_SERVER
+    };
+  }
+
+  // GetMCPServersForSecret implements the get MCP servers for secret method which lists the MCP servers
+  // in the Redpanda cluster for the given secret.
+  rpc GetMCPServersForSecret(GetMCPServersForSecretRequest) returns (GetMCPServersForSecretResponse) {
+    option (google.api.http) = {get: "/v1/redpanda-connect/mcp-servers-for-secret"};
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+      summary: "Get Redpanda Connect MCP servers for secret"
+      description: "Get Redpanda Connect MCP servers for a given secret."
+      responses: {
+        key: "200"
+        value: {
+          description: "OK"
+          schema: {
+            json_schema: {ref: ".redpanda.api.dataplane.v1.GetMCPServersForSecretResponse"}
+          }
+        }
+      }
+    };
+    option (redpanda.api.auth.v1.authorization) = {
+      required_permission: PERMISSION_VIEW
+      api: API_REDPANDA_MCP_SERVER
+    };
+  }
+
+  // GetMCPServersBySecrets implements the get MCP servers by secrets method which lists the MCP servers
+  // in the Redpanda cluster for all secrets.
+  rpc GetMCPServersBySecrets(GetMCPServersBySecretsRequest) returns (GetMCPServersBySecretsResponse) {
```

Comment:
do we need this one? also, maybe we can rather integrate these into List, instead of having multiple GetBy*, using the filter field.

Reply:
Moved into List